docs/eks.md

```diff
@@ -0,0 +1,309 @@
+`AMI_ID` should be an AL2023-based [EKS optimized AMI](https://docs.aws.amazon.com/eks/latest/userguide/eks-optimized-ami.html). This setup relies on [`nodeadm`](https://awslabs.github.io/amazon-eks-ami/nodeadm/) to configure containerd and the Kublet which is not available in AL2-based AMIs.  
+
+```
+AWS_REGION=us-west-2
+CLUSTER_NAME=soci
+KUBERNETES_VERSION=1.30
+ARCH=x86_64
+INSTANCE_TYPE=t3.large
+AMI_ID=$(aws ssm get-parameter --name /aws/service/eks/optimized-ami/${KUBERNETES_VERSION}/amazon-linux-2023/${ARCH}/standard/recommended/image_id  --region $AWS_REGION --query "Parameter.Value" --output text)
+```
+
+### Step 2: Create an EKS cluster 
+
+Next we will create a cluster using eksctl. This step will create a cluster and the necessary resources for EKS to function, but importantly will not create any nodegroups. We will create nodegroups later using a custom launch template that installs and configures the SOCI snapshotter.
+
+```
+eksctl create cluster \
+  --without-nodegroup \
+  --name $CLUSTER_NAME \
+  --version $KUBERNETES_VERSION \
+  --region $AWS_REGION
+```
+
+### Step 3: Create Node configuration file
+
+This config will allow our node to join the cluster with updated containerd/kubelet config to support SOCI. 
+
+```
+CLUSTER_ENDPOINT=$(aws eks describe-cluster \
+  --name $CLUSTER_NAME \
+  --region $AWS_REGION \
+  --output text \
+  --query 'cluster.endpoint')
+CLUSTER_CERTIFICATE_AUTHORITY=$(aws eks describe-cluster \
+  --name $CLUSTER_NAME --region $AWS_REGION \
+  --output text \
+  --query 'cluster.certificateAuthority.data')
+CLUSTER_CIDR=$(aws eks describe-cluster \
+  --name $CLUSTER_NAME \
+  --region $AWS_REGION \
+  --output text \
+  --query 'cluster.kubernetesNetworkConfig.serviceIpv4Cidr')
+
+cat <<EOF > node_config.yaml
+apiVersion: node.eks.aws/v1alpha1
+kind: NodeConfig
+spec:
+  cluster:
+    name: $CLUSTER_NAME
+    apiServerEndpoint: $CLUSTER_ENDPOINT
+    certificateAuthority: $CLUSTER_CERTIFICATE_AUTHORITY
+    cidr: $CLUSTER_CIDR
+  kubelet:
+    config:
+      imageServiceEndpoint: unix:///run/soci-snapshotter-grpc/soci-snapshotter-grpc.sock
+  containerd: 
+    config: |
+      [proxy_plugins.soci]
+        type = "snapshot"
+        address = "/run/soci-snapshotter-grpc/soci-snapshotter-grpc.sock"
+        [proxy_plugins.soci.exports]
+          root = "/var/lib/soci-snapshotter-grpc"
+      [plugins."io.containerd.grpc.v1.cri".containerd]
+        snapshotter = "soci"
+        # This line is required for containerd to send information about how to lazily load the image to the snapshotter
+        disable_snapshot_annotations = false
+EOF
+```
+
+
+### Step 4: Create SOCI install script
+
+This will create an script that will be run on instance boot to install and configure the SOCI snapshotter.
+
+```
+cat <<'EOF_SCRIPT' >install_soci.sh
+#!/bin/bash
+# Set environment variables
+ARCH=$(uname -m | sed s/aarch64/arm64/ | sed s/x86_64/amd64/)
+VERSION=0.7.0
```

Comment:
Do you have thoughts if this should be updated on each release?

Reply:
Ideally, yes.